packages/welldone-wallet/src/lib/welldone.ts

```diff
@@ -295,6 +331,15 @@ const WelldoneWallet: WalletBehaviourFactory<InjectedWallet> = async ({
       };
     },
 
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async signMessage(
+      message: Uint8Array | SignMessageParams,
+      accountId?: string
+    ) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return (await signer.signMessage(message as any, accountId)) as any;
+    },
```

Comment:
As mentioned in the comment: https://github.com/near/wallet-selector/issues/941#issuecomment-1730921716

This PR could be as simple as adding the signMessage this way, and to all the signing in the wallet

```ts
async signMessage({ message, nonce, recipient }) {
  return await _state.wallet.request("near", {
    method: "dapp:signMessage",
    params: [{ message, nonce, recipient }],
  });
},
```

The user does not need to be signed-in to wallet-selector to signMessage, this means dApp developers can directly call the `signMessage` without first calling the `signIn` method of wallet-selector.

```ts
const welldone = await selector.wallet("welldone-wallet");

await welldone.signMessage({
    message: "test message",
    nonce: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,24,26,27,28,29,30,31],
    recipient: "www.google.com",
});
```

Reply:
Thanks for the feedback. I committed the fix :)

Just double-checking because of the example you provided. Do you suggest NEP413 serializing inside the wallet? Instead we implemented it in the wallet-selector which is outside the wallet.

Reply:
> Just double-checking because of the example you provided. Do you suggest NEP413 serializing inside the wallet? Instead we implemented it in the wallet-selector which is outside the wallet.

Yes, ideally it would be nice if the logic for serializing is kept in the wallet.

Reply:
Understood, the serialization can be kept as it is now 👍 .